### PR TITLE
[14.0][IMP] shopfloor_reception_mobile: Make states hookable

### PR DESCRIPTION
**Pull request by mt-software-de**

### CODE REVIEW

**Comment by mt-software-de** [comment]
@simahawk could you please have a look

File: shopfloor_reception_mobile/static/src/scenario/reception_states.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+/*
+    Define states for reception scenario.
+    @param $instance VueJS component instance
+*/
+export const reception_states = function ($instance) {
+    return {
+        init: {
+            enter: () => {
+                this.wait_call(this.odoo.call("start"));
```
Comment (simahawk) [inline shopfloor_reception_mobile/static/src/scenario/reception_states.js:15]: if you don't bind `this` to `$instance` you have to replace all this `this` w/ `$instance`.

File: shopfloor_reception_mobile/static/src/scenario/reception.js [hunk 2]
```diff
@@ -454,295 +456,17 @@ const Reception = {
             }
             return picking.origin.includes(input);
         },
+        _get_states: function () {
+            return reception_states(this);
```
Comment (simahawk) [inline shopfloor_reception_mobile/static/src/scenario/reception.js]: can you try this? See comment below.
```suggestion
            return reception_states.bind(this)();
```

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopfloor_reception_mobile/static/src/scenario/reception_states.js (hunk 1 above)
Comment (mt-software-de) [reply shopfloor_reception_mobile/static/src/scenario/reception_states.js:15]: oh yes i forget to replace it

File: shopfloor_reception_mobile/static/src/scenario/reception.js (hunk 2 above)
Comment (simahawk) [reply shopfloor_reception_mobile/static/src/scenario/reception.js]: Out of curiosity: have you tried this? If it works as I expect, is probably more clear.
Not a blocker though, as we've done the same for checkout_states.

Comment (mt-software-de) [reply shopfloor_reception_mobile/static/src/scenario/reception.js]: i tried it this should work

**Review by simahawk — APPROVED** [review]
LG, thanks

**Review by sebalix — APPROVED** [review]

**Pull request merged**